lib/adapters/apply-edit-adapter.ts

```diff
@@ -81,6 +90,86 @@ export default class ApplyEditAdapter {
     return { applied }
   }
 
+  private static async handleResourceOperation(edit: (CreateFile | RenameFile | DeleteFile)): Promise<void> {
+    if (DeleteFile.is(edit)) {
+      const path = Convert.uriToPath(edit.uri)
+      const exists = fs.existsSync(path)
```

Comment:
We should use async fs functions to prevent blocking the main thread.

Reply:
Replaced by:
```typescript
const exists = await fs.promises.stat(path).then(() => true).catch(() => false)
```

Reply:
I think `fs.promises.access` would be better since we don't use the stat object.

Reply:
There is also the path-exits package that exactly does this.
https://www.npmjs.com/package/path-exists

Reply:
seems a little much to install a dependency for a one liner

Reply:
I agree with @UziTech 
I already find it annoying having to require `rimraf`;)